### PR TITLE
Create MainWindowWithProgressDialogs

### PR DESCRIPTION
**Pull request by lauramurgatroyd**
Pulls out some functionality from the SessionMainWindow into the MainWindowWithProgressDialogs. The SessionMainWindow now inherits from MainWindowWithProgressDialogs

The MainWindowWithProgressDialogs is  a base class for a main window, with a menu bar, and ability to create ProgressTimerDialogs.

Created this as in some apps e.g. standalone viewer, may want progress bars but not session creation.

Also renamed SessionMainWindow -> MainWindowWithSessionManagement

### CODE REVIEW

**Review by paskino — CHANGES_REQUESTED** [review]

File: eqt/ui/ProgressMainWindow.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        '''
+        You may wish to override this in a child class, to add
+        additional menu options.
+
+        The menu bar is available as self.menu_bar, and the menus are
+        available as self.menus, which is a dictionary of the menu names
+        and the corresponding QMenu objects.
+        '''
+
+        pass
```
[the remark this answers is not in the document]
Comment (lauramurgatroyd) [reply eqt/ui/ProgressMainWindow.py:107]: Yes, in the docstring it tells the user to override this in a child class

File: eqt/ui/ProgressMainWindow.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            organisation_name = app_name
+
+        self.settings = QSettings(organisation_name, settings_name)
+
+        self.setAppStyle()
+
+        self.createMenu()
+        self.addToMenu()
+
+        self.progress_windows = {}
```
[the remark this answers is not in the document]
Comment (lauramurgatroyd) [reply eqt/ui/ProgressMainWindow.py]: done

**Comment by lauramurgatroyd** [comment]
Updated to make progress_windows a property and renamed the classes to:
- MainWindowWithSessionManagement
- MainWindowWithProgressDialogs

**Comment by lauramurgatroyd** [comment]
Removed addToMenu method

**Review by paskino — APPROVED** [review]

**Pull request merged**